airflow/providers/google/cloud/hooks/dataflow.py

```diff
@@ -622,7 +622,7 @@ def start_template_dataflow(
         dataflow_template: str,
         project_id: str,
         append_job_name: bool = True,
-        on_new_job_id_callback: Optional[Callable[[str], None]] = None,
+        on_new_job_callback: Optional[Callable[[dict], None]] = None,
```

Comment:
It looks like a breaking change. Am I right?

Reply:
@mik-laj A little, We need the information like 'project_id' and 'location' from Job not from the config. In the operator we have I think 3 or 4 possibilieties to define that properties, so If we want kill correct job in the correct location we should use properties from Job.

Reply:
How about  adding new callback instead? It feels a bit risky, and it does not hurt to have two parameters (we can deprecate the old one and raise deprecation warning)